TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetAllSpecialismsHandler.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.PowerPlatform.Dataverse.Client;
+using Microsoft.Xrm.Sdk.Messages;
+using Microsoft.Xrm.Sdk.Query;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.Core.Dqt.QueryHandlers;
+
+public class GetAllSpecialismsHandler : ICrmQueryHandler<GetAllSpecialismsQuery, dfeta_specialism[]>
+{
+    public async Task<dfeta_specialism[]> Execute(GetAllSpecialismsQuery query, IOrganizationServiceAsync organizationService)
+    {
+        var filter = new FilterExpression(LogicalOperator.And);
+        filter.AddCondition(dfeta_specialism.Fields.StateCode, ConditionOperator.Equal, (int)dfeta_specialismState.Active);
+        var valueFilter = new FilterExpression(LogicalOperator.Or);
+        valueFilter.AddCondition(dfeta_specialism.Fields.dfeta_Value, ConditionOperator.Equal, "Hearing");
+        valueFilter.AddCondition(dfeta_specialism.Fields.dfeta_Value, ConditionOperator.Equal, "Multi-Sensory");
+        valueFilter.AddCondition(dfeta_specialism.Fields.dfeta_Value, ConditionOperator.Equal, "Visual");
```

Comment:
I think there's an `In` operator that might make this a little neater

Reply:
Done